src/mapper/pkg/resolvers/helpers.go

```diff
@@ -99,10 +100,18 @@ func (i *IntentsHolder) AddIntent(srcService model.OtterizeServiceIdentity, dstS
 		intents = goset.NewSet[model.OtterizeServiceIdentity]()
 		i.store[srcService] = intents
 	}
-	intents.Add(model.OtterizeServiceIdentity{Name: dstService.Name, Namespace: namespace})
+	intent := model.OtterizeServiceIdentity{Name: dstService.Name, Namespace: namespace}
+	if !i.newIntentFound && !intents.Contains(intent) {
+		i.newIntentFound = true
+	}
+	intents.Add(intent)
+}
+
+func (i *IntentsHolder) NewIntentsExists() bool {
+	return i.newIntentFound
 }
 
-func (i *IntentsHolder) GetIntentsPerService(namespaces []string) map[model.OtterizeServiceIdentity][]model.OtterizeServiceIdentity {
+func (i *IntentsHolder) GetIntentsPerService(namespaces []string, markAsRead bool) map[model.OtterizeServiceIdentity][]model.OtterizeServiceIdentity {
```

Comment:
Ummm it's a bit inconvenient to need to fill this parameter when all you want to do is read the intents.